src/pages/home/report/comment/TextWithEmojiFragment.tsx

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import type {StyleProp, TextStyle} from 'react-native';
+import Text from '@components/Text';
+import useLocalize from '@hooks/useLocalize';
+import useTheme from '@hooks/useTheme';
+import useThemeStyles from '@hooks/useThemeStyles';
+import * as DeviceCapabilities from '@libs/DeviceCapabilities';
+import * as EmojiUtils from '@libs/EmojiUtils';
+import variables from '@styles/variables';
+import CONST from '@src/CONST';
+
+type TextWithEmojiFragmentProps = {
+    /** The message to be displayed */
+    message: string;
+
+    /** Additional styles to add after local styles. */
+    passedStyles?: StyleProp<TextStyle>;
+
+    /** Should this message fragment be styled as deleted? */
+    styleAsDeleted?: boolean;
+
+    /** Should this message fragment be styled as muted? */
+    styleAsMuted?: boolean;
+
+    /** Is message displayed on narrow screen? */
+    isSmallScreenWidth?: boolean;
+
+    /** Should "(edited)" suffix be rendered? */
+    isEdited?: boolean;
+
+    /** Does message contain only emojis? */
+    emojisOnly?: boolean;
+};
+
+function TextWithEmojiFragment({message, passedStyles, styleAsDeleted, styleAsMuted, isSmallScreenWidth, isEdited, emojisOnly}: TextWithEmojiFragmentProps) {
```

Comment:
Let's `useWindowDimensions` in the component rather than passing `isSmallScreenWidth` as a prop.